201-vmss-automation-dsc/nested/provisionServer.json

```diff
@@ -0,0 +1,254 @@
+                  "type": "DSC",
+                  "typeHandlerVersion": "2.76",
+                  "autoUpgradeMinorVersion": true,
+                  "protectedSettings": {
+                    "Items": {
+                      "registrationKeyPrivate": "[listKeys(resourceId('Microsoft.Automation/automationAccounts/', parameters('automationAccountName')), '2018-01-15').Keys[0].value]"
+                    }
+                  },
+                  "settings": {
+                    "Properties": [
+                      {
+                        "Name": "RegistrationKey",
+                        "Value": {
+                          "UserName": "PLACEHOLDER_DONOTUSE",
+                          "Password": "PrivateSettingsRef:registrationKeyPrivate"
+                        },
+                        "TypeName": "System.Management.Automation.PSCredential"
+                      },
+                      {
+                        "Name": "RegistrationUrl",
+                        "Value": "[reference(concat('Microsoft.Automation/automationAccounts/', parameters('automationAccountName'))).registrationUrl]",
+                        "TypeName": "System.String"
+                      },
+                      {
+                        "Name": "NodeConfigurationName",
+                        "Value": "[parameters('nodeConfigurationName')]",
+                        "TypeName": "System.String"
+                      },
+                      {
+                        "Name": "ConfigurationMode",
+                        "Value": "ApplyandAutoCorrect",
+                        "TypeName": "System.String"
+                      },
+                      {
+                        "Name": "RebootNodeIfNeeded",
+                        "Value": true,
+                        "TypeName": "System.Boolean"
+                      },
+                      {
+                        "Name": "ActionAfterReboot",
+                        "Value": "ContinueConfiguration",
+                        "TypeName": "System.String"
+                      }
+                    ]
+                  }
+                }
+              }
+            ]
+          },
+          "osProfile": {
+            "computerNamePrefix": "[toLower(parameters('vmssName'))]",
+            "adminUsername": "[parameters('adminUsername')]",
+            "adminPassword": "[parameters('adminPassword')]"
+          },
+          "networkProfile": {
+            "networkInterfaceConfigurations": [
+              {
+                "name": "[parameters('nicName')]",
+                "properties": {
+                  "primary": true,
+                  "ipConfigurations": [
+                    {
+                      "name": "ipConfig",
+                      "properties": {
+                        "subnet": {
+                          "id": "[concat(resourceId('Microsoft.Network/virtualNetworks/',parameters('virtualNetworkName')),'/subnets/',parameters('subnetName'))]"
```

Comment:
remove concat, use resourceId - here and below

Reply:
same as above

Reply:
@bmoore-msft

Reply:
@bmoore-msft I think this is the last feedback item to address.